attractions/dropdown/dropdown.svelte

```diff
@@ -13,9 +37,78 @@
    * @type {boolean}
    */
   export let top = false;
+  /**
+   * Dropdown horizontal alignment relative to the shell.
+   * @type {string}
+   */
+  export let horizontalAlignment = HorizontalAlignment.autoStart;
+  /**
+   * Dropdown vertical alignment relative to the shell.
+   * @type {string}
+   */
+  export let verticalAlignment = VerticalAlignment.autoBottom;
+
+  const isDropdownOpen = getContext(CONTEXT_IS_DROPDOWN_OPEN);
+  const getDropdownShellBoundary = getContext(
+    CONTEXT_GET_DROPDOWN_SHELL_BOUNDARY
+  );
+
+  let dropdownElement, isVerticalAlignTop, isHorizontalAlignEnd;
+
+  function getIsVerticalAlignTop(dropdownBound, dropdownShellBound) {
+    const { height } = dropdownBound;
+    const { top, bottom } = dropdownShellBound;
+
+    switch (verticalAlignment) {
+      case VerticalAlignment.top:
+        return true;
+      case VerticalAlignment.bottom:
+        return false;
+      case VerticalAlignment.autoTop:
+        return height <= top;
+      default:
+        // auto-bottom by default
+        return height > window.innerHeight - bottom;
+    }
+  }
+
+  function getIsHorizontalAlignEnd(dropdownBound, dropdownShellBound) {
+    const { width } = dropdownBound;
+    const { left, right } = dropdownShellBound;
+
+    switch (horizontalAlignment) {
+      case HorizontalAlignment.end:
+        return true;
+      case HorizontalAlignment.start:
+        return false;
+      case HorizontalAlignment.autoEnd:
+        return width <= left;
+      default:
+        // auto-start by default
+        return width > window.innerWidth - right;
```

Comment:
I think this should be counted from the left, because when the dropdown is start-aligned, it spans the entire width of the dropdown shell and the rest of the page on the right.
```suggestion
        return width > window.innerWidth - left;
```